PlaybackLab/DownloadToOwnSampleApp/DownloadToOwnSampleApp/AppDelegate.m

```diff
@@ -7,48 +7,43 @@
 //
 
 #import "AppDelegate.h"
-
-@interface AppDelegate ()
-
-@end
+#import <OoyalaSDK/OODtoAsset.h>
+#import "OptionsDataSource.h"
 
 @implementation AppDelegate
 
-
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
   // Override point for customization after application launch.
+  [OODtoAsset restoreAllDownloadsForDtoAssets:OptionsDataSource.dtoAssets];
```

Comment:
Please check if using method `[OODtoAsset restoreAllDownloadsForDtoAssets:OptionsDataSource.dtoAssets];` in `-application:didFinishLaunchingWithOptions:` is correct? Because this appdelegate method called only once, during launching, but need to be called after each background->foreground cycle. So my suggestion is to move into `-applicationWillEnterForeground:`.

Reply:
This call was made for another purpose: to restore sessions after app was suspended or terminated by the system.
Moving to `applicationWillEnterForeground:` won't change its behavior.

Reply:
Actually, I've rechecked once again and I can confirm that going to background and switching back should not have any impact on the delegates, however I did observer the same behavior before.
![Screen Shot 2019-05-22 at 12 01 50](https://user-images.githubusercontent.com/10694577/58161776-91345300-7c89-11e9-8f0b-d917803ed511.png)
The screenshot indicates that switching between foreground/background leaves the delegate the same: an instance of `OOAssetDownloadManager`. The mentioned method `URLSession:assetDownloadTask:didLoadTimeRange:totalTimeRangesLoaded:timeRangeExpectedToLoad:` is a delegate method and sometimes it fails to deliver all the messages.

Reply:
So what is the our resolution about 
_after pushing app into background and returning into foreground progress bar for downloading is stuck on some %._ Is it problem that must be resolved? Or it can be left as known issue?

Reply:
I'd say that it's not the problem in our SDK or sample app implementation.

Reply:
Ok